javascript/attributes.js

```diff
@@ -31,23 +33,52 @@ export const extractElementAttributes = element => {
       return memo
     }, {})
 
-  attrs.value = element.value
   attrs.checked = !!element.checked
   attrs.selected = !!element.selected
   attrs.tag_name = element.tagName
-  if (element.tagName.match(/select/i)) {
-    if (element.multiple) {
-      const checkedOptions = Array.prototype.slice.call(
-        element.querySelectorAll('option:checked')
-      )
-      attrs.values = checkedOptions.map(o => o.value)
-    } else if (element.selectedIndex > -1) {
-      attrs.value = element.options[element.selectedIndex].value
+
+  if (hasMultipleOptions(element)) {
+    attrs.values = collectCheckedOptions(element)
+    delete attrs.value
+  } else {
+    attrs.value = element.value
+    if (element.tagName.match(/select/i)) {
+      if (element.selectedIndex > -1) {
+        attrs.value = element.options[element.selectedIndex].value
+      }
     }
   }
+
   return attrs
 }
 
+const hasMultipleOptions = element => {
+  const multipleElementCount = document.querySelectorAll(
+    `input[type="${element.type}"][name="${element.name}"]`
+  ).length
+
+  return (
+    (element.tagName.match(/select/i) && element.multiple) ||
+    multipleElementCount > 1
+  )
+}
```

Comment:
We could simplify or eliminate this guard function if we adopt my proposal above as all we'd need to verify is that we're dealing with a select.

Reply:
I'm not sure I'm following. What about multiple `input`s ?

Reply:
Something like this is what I was thinking when I wrote that comment. Not sure if it covers all edge cases though.

```js
const hasMultipleOptions = element => {
  if (element.tagName === 'SELECT' && element.multiple) return true 
  if (element.tagName === 'CHECKBOX' && document.querySelectorAll(`[name="${element.name}"]`).length > 1)
    return true
  return false
}
```

Reply:
I see... well I wanted it to be a bit more generic because I'm not 100% sure that checkboxes just cover all scenarios... there might be some exotic cases, as you said, of other multiple elements...

Reply:
but I'll rework if you think it's necessary